chore: add Js string methods transcript (#58897)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-working-with-common-string-methods/67326c29dcd98fc5ecc49779.md

# --description--

Watch the video or read the transcript and answer the questions below.

# --transcript--

How can you replace parts of a string with another?

In JavaScript, there are many scenarios where you may need to replace a portion of a string with another string.

For instance, you might need to update user information in a URL, change the formatting of dates, or correct errors in user-generated content.

The `replace()` method in JavaScript allows you to find a specified value (like a word or character) in a string and replace it with another value.

Here is the basic syntax:

```js
string.replace(searchValue, newValue);
```

> `searchValue` is the value you want to search for in the string. It can be either a string or a regular expression (regex), which describes patterns in text. This allows you to search for and manipulate strings in a flexible and powerful way. You’ll learn more about regular expressions in future lecture videos.

Reviewer: ```suggestion
`searchValue` is the value you want to search for in the string. It can be either a string or a regular expression (regex), which describes patterns in text. This allows you to search for and manipulate strings in a flexible and powerful way. You'll learn more about regular expressions in future lecture videos.
```

> The `newValue` is the value that will replace the `searchValue`.
> 
> Here’s a simple example:

Reviewer: ```suggestion
The `newValue` is the value that will replace the `searchValue`. Here's a simple example:
```

```js
let text = "I love JavaScript!";
let newText = text.replace("JavaScript", "coding");

console.log(newText);  // Output: "I love coding!"
```

> In this example, the word `"JavaScript"` is found within the string and is replaced with `"coding"`.

Reviewer: ```suggestion
In this example, the word `JavaScript` is found within the string and is replaced with `coding`.
```

> The `replace()` method is case-sensitive, meaning that it will only find exact matches of the `searchValue`. 
> 
> For example:

Reviewer: ```suggestion
The `replace()` method is case-sensitive, meaning that it will only find exact matches of the `searchValue`. For example:
```

```js
let sentence = "I enjoy working with JavaScript.";
let updatedSentence = sentence.replace("javascript", "coding");

console.log(updatedSentence);  // Output: "I enjoy working with JavaScript."
```

> Here, since `"javascript"` (with lowercase "j") does not match `"JavaScript"` (with uppercase "J"), the replacement is not made.

Reviewer: ```suggestion
Here, since `javascript` (with lowercase `j`) does not match `JavaScript` (with uppercase `J`), the replacement is not made.
```

By default, the `replace()` method will only replace the first occurrence of the `searchValue`. If the value appears multiple times in the string, only the first one will be replaced:

```js
let phrase = "Hello, world! Welcome to the world of coding.";
let updatedPhrase = phrase.replace("world", "universe");

console.log(updatedPhrase);  // Output: "Hello, universe! Welcome to the world of coding."
```

> Notice that only the first occurrence of `"world"` is replaced with `"universe"`.

Reviewer: ```suggestion
Notice that only the first occurrence of `world` is replaced with `universe`.
```

The `replace()` method in JavaScript is a powerful and flexible tool for string manipulation.

It lets you replace specific parts of a string, whether you're dealing with individual characters, words, or complex patterns using regular expressions.

> While it’s ideal for straightforward replacements, understanding its case sensitivity and default behavior (like replacing only the first occurrence) can help you use it more effectively.

Reviewer: ```suggestion
While it's ideal for straightforward replacements, understanding its case sensitivity and default behavior (like replacing only the first occurrence) can help you use it more effectively.
```
